### PR TITLE
Update templates algo

### PR DESCRIPTION
**Pull request by nbirillo**

### CODE REVIEW

File: templates/src/templates/diffs/filter_by_diff.py [hunk 1]
```diff
@@ -64,11 +63,68 @@ def issues_offsets_to_positions(offsets: List[int], code_lines: List[str]) -> Li
     return issues_positions
 
 
+def to_cleanup_semantic(template_lines: List[str], code_lines: List[str]):
```
Comment (tiginamaria) [inline templates/src/templates/diffs/filter_by_diff.py]: Maybe add return type  `-> bool:`

**Review by tiginamaria — APPROVED** [review]
All is great for now, but I still not sure everything is ok with patch merging using `matcher.diff_cleanupSemantic`. We need to run it in big data and see that all template issues are detected. Maybe we need to think about applying `matcher.diff_cleanupSemantic` only on part of code with comment in future?

File: templates/src/templates/diffs/filter_by_diff.py (hunk 1 above)
Comment (nbirillo) [reply templates/src/templates/diffs/filter_by_diff.py]: great idea, thanks!

**Comment by nbirillo** [comment]
I will try to apply diff_cleanupSemantic onty to comments parts in the next pr

**Pull request merged**